cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractReactiveLimitsOwnerConversion.java

```diff
@@ -32,49 +36,89 @@ protected void convertReactiveLimits(Generator g) {
         if (p.containsKey("ReactiveCapabilityCurve")) {
             String curveId = p.getId("ReactiveCapabilityCurve");
-            ReactiveCapabilityCurveAdder rcca = g.newReactiveCapabilityCurve();
             PropertyBags curveData = context.reactiveCapabilityCurveData(curveId);
-            curveData.forEach(d -> {
+            if (curveData.size() >= 2) {
+
+                Map<Double, Range<Double>> qRanges = new HashMap<>();
+                curveData.forEach(d -> {
+                    double p = d.asDouble("xvalue");
+                    double minQ = d.asDouble("y1value");
+                    double maxQ = d.asDouble("y2value");
+
+                    if (!checkPointValidity(p, minQ, maxQ)) {
+                        return;
+                    }
+
+                    Range<Double> qRange = fixedMinMaxQ("ReactiveCapabilityCurve Point", minQ, maxQ);
+
+                    // check if there is a previous point with same p
+                    Range<Double> prev = qRanges.get(p);
+                    if (prev != null) {
+                        if (prev.isConnected(qRange)) {
+                            fixed("reactive capability curve", String.format("point merged with another one with same p (%f)", p));
+                            qRanges.put(p, prev.span(qRange));
+                        } else {
+                            ignored("reactive capability curve point", String.format("another point with same p (%f) and a disconnected reactive range", p));
+                        }
+                    } else {
+                        qRanges.put(p, qRange);
+                    }
+                });
+
+                // create curve
+                ReactiveCapabilityCurveAdder rcca = g.newReactiveCapabilityCurve();
+                for (Map.Entry<Double, Range<Double>> e : qRanges.entrySet()) {
+                    double p = e.getKey();
+                    Range<Double> qRange = e.getValue();
+                    rcca.beginPoint()
+                            .setP(p)
+                            .setMinQ(qRange.lowerEndpoint())
+                            .setMaxQ(qRange.upperEndpoint())
+                            .endPoint();
+                }
+                rcca.add();
+            } else if (curveData.size() == 1) {
+                fixed("reactive capability curve", "only one point");
+                PropertyBag d = curveData.get(0);
                 double p = d.asDouble("xvalue");
                 double minQ = d.asDouble("y1value");
                 double maxQ = d.asDouble("y2value");
-                // Check the point is valid
-                if (Double.isNaN(p) || Double.isNaN(minQ) || Double.isNaN(maxQ)) {
-                    String reason = String.format("Incomplete point p, minQ, maxQ = %f, %f, %f", p, minQ, maxQ);
-                    ignored("ReactiveCapabilityCurvePoint", reason);
-                    return;
+                if (checkPointValidity(p, minQ, maxQ)) {
```

Comment:
if this point is not valid, we should still try to read `minQ` and `maxQ` in the traditional way.

Reply:
Fixed in the new algo, check on point count is done after removing duplicate points.